src/abbfreeathome/devices/trigger.py

```diff
@@ -0,0 +1,56 @@
+"""Free@Home Trigger Class."""
+
+import logging
+from typing import Any
+
+from ..api import FreeAtHomeApi
+from ..bin.pairing import Pairing
+from .base import Base
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Trigger(Base):
+    """Free@Home Trigger Class."""
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        channel_id: str,
+        channel_name: str,
+        inputs: dict[str, dict[str, Any]],
+        outputs: dict[str, dict[str, Any]],
+        parameters: dict[str, dict[str, Any]],
+        api: FreeAtHomeApi,
+        floor_name: str | None = None,
+        room_name: str | None = None,
+    ) -> None:
+        """Initialize the Free@Home Trigger class."""
+        super().__init__(
+            device_id,
+            device_name,
+            channel_id,
+            channel_name,
+            inputs,
+            outputs,
+            parameters,
+            api,
+            floor_name,
+            room_name,
+        )
+
+    async def press(self):
+        """Press the trigger."""
+        await self._set_trigger_datapoint()
+
+    async def _set_trigger_datapoint(self):
+        _trigger_input_id, _trigger_input_value = self.get_input_by_pairing(
+            pairing=Pairing.AL_TIMED_START_STOP
```

Comment:
I'm actually unsure because I don't have a device like this, why use `AL_TIMED_START_STOP` instead of `AL_SCENE_CONTROL`?

Reply:
As this device has no state I would also say it is a button in HA

Reply:
Well, when I set AL_TIMED_START_STOP to 1 the trigger gets activated :smile:

Reply:
You just have formatting issues, otherwise this should be good to merge.

Reply:
I don't understand where these formatting issues should come from. If I look at the output of the action it states, that freeathome.py "would be reformatted". What does that means?

Reply:
The utility `ruff` will reformat the document in a consistant way, it uses black for most of it's formatting. You can read more about ruff and it's formatter here: https://docs.astral.sh/ruff/formatter/

The GitHub action is testing ruff to see if it would make changes, if it would then the action fails. You can install and run `ruff format`.

To make this easier, if you use Visual Studio you can choose to automatically format files on save, or you can format document from the command pallete.

![Screenshot 2024-10-15 at 14 52 27](https://github.com/user-attachments/assets/3ef25542-46dc-4149-9c7c-30a04535ec8c)